core/src/main/java/com/scalar/db/api/DistributedStorageFactory.java

```diff
@@ -0,0 +1,35 @@
+package com.scalar.db.api;
+
+import com.scalar.db.config.DatabaseConfig;
+
+/**
+ * A factory class that creates {@link DistributedStorage} and {@link DistributedStorageAdmin}
+ * instances. Each storage adapter should implement this class to instantiate its implementations of
+ * {@link DistributedStorage} and {@link DistributedStorageAdmin}. The implementations are assumed
+ * to be loaded by {@link java.util.ServiceLoader}.
+ */
+public interface DistributedStorageFactory {
```

Comment:
Introduced `DistributedStorageFactory` and added implementations of each storage adapter.